opnode/l2/driver.go

```diff
@@ -0,0 +1,207 @@
+package l2
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/event"
+
+	"github.com/ethereum-optimism/optimistic-specs/opnode/eth"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type EngineDriver struct {
+	Ctx context.Context
```

Comment:
I need to understand how this is being used before making a final recommendation, but generally people try to avoid embedding contexts in structs (though there is some disagreement)
https://go.dev/blog/context-and-structs
https://github.com/golang/go/issues/22602

Reply:
The idea is that all background work, like calls to the engine API, can wrap this context, so they don't linger around when we shut down / replace an engine. When we shut down the engine we close this context. (maybe we should just make it a private field, and cancel the context when we `Close` the driver.

Reply:
Alternatively, we can remove it from the struct, and add it to the individual methods. Think I like that better now that you mention it

Reply:
Fixed in 153473aac2f602a0e5e7e524011d7f726ae84726